cardinal/router/router.go

```diff
@@ -60,16 +61,17 @@ type router struct {
 	// serverAddr is the address the evmServer listens on. This is set once `Start` is called.
 	serverAddr string
 	port       string
+	routerKey  string
 }
 
-func (r *router) TransactionIterator() iterator.Iterator {
-	return iterator.New(r.provider.GetMessageByID, r.namespace, r.ShardQuerier)
-}
+func New(namespace, sequencerAddr, baseShardQueryAddr, routerKey string, provider Provider) (Router, error) {
+	rtr := &router{namespace: namespace, port: defaultPort, provider: provider, routerKey: routerKey}
 
-func New(namespace, sequencerAddr, baseShardQueryAddr string, provider Provider) (Router, error) {
-	rtr := &router{namespace: namespace, port: defaultPort, provider: provider}
-
-	conn, err := grpc.Dial(sequencerAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	conn, err := grpc.Dial(
+		sequencerAddr,
+		grpc.WithUnaryInterceptor(rtr.clientCallInterceptor),
```

Comment:
Use `grpc.WithPerRPCCredentials` instead. 

As mentioned above, we should have a package that is both used by EVM and Cardinal with a router key struct type that implements the `grpc.PerRPCCredentials` interface (https://pkg.go.dev/google.golang.org/grpc/credentials#PerRPCCredentials).

This avoids hardcoding the metadata key individually that might become a footgun when we need to make a change to that in the fture.

Example: https://github.com/kelseyhightower/grpc-hello-service/blob/master/credentials/jwt/jwt.go

Reply:
done